mlir/test/Dialect/Linalg/hoisting.mlir

```diff
@@ -308,6 +308,61 @@ module attributes {transform.with_named_sequence} {
 
 // -----
 
+// CHECK-LABEL:  func.func @no_hoisting_zero_trip_loop
+func.func @no_hoisting_zero_trip_loop(%arg0: memref<20xi32>, %lb: index, %ub: index) {
```

Comment:
You are testing 3 different cases in this one function. Could you instead split this into 3 separate, self-contained functions? IMHO, that would be more in line with the [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices):

> FileCheck tests should be as self-contained as possible and focus on testing the minimal set of functionalities needed.

In fact, other tests in this file also focus on testing one-loop nest per function.